tests/test_endpoint_correction.py

```diff
@@ -0,0 +1,226 @@
+    def u_lhs_fn(x_t):
+        return bond_fn(x_t) + angle_fn(x_t) + core_restr(x_t)
+
+    # right hand state is post-processed from independent gas phase simulations
+    def u_rhs_fn(x_t):
+        return bond_fn(x_t) + angle_fn(x_t)
+
+    return u_lhs_fn, u_rhs_fn, core_idxs, core_params, mol_a, mol_b
+
+
+# (ytz): do not remove, useful for visualization in pymol
+def make_conformer(mol_a, mol_b, conf_c):
+    mol_a = Chem.Mol(mol_a)
+    mol_b = Chem.Mol(mol_b)
+
+    """Remove all of mol's conformers, make a new mol containing two copies of mol,
+    assign positions to each copy using conf_a and conf_b, respectively, assumed in nanometers"""
+    assert conf_c.shape[0] == mol_a.GetNumAtoms() + mol_b.GetNumAtoms()
+    mol_a.RemoveAllConformers()
+    mol_b.RemoveAllConformers()
+    mol = Chem.CombineMols(mol_a, mol_b)
+    cc = Chem.Conformer(mol.GetNumAtoms())
+    conf = np.copy(conf_c)
+    conf *= 10  # TODO: label this unit conversion?
+    for idx, pos in enumerate(np.asarray(conf)):
+        cc.SetAtomPosition(idx, (float(pos[0]), float(pos[1]), float(pos[2])))
+    mol.AddConformer(cc)
+
+    return mol
+
+
+def test_endpoint_correction():
+
+    # this PR tests that endpoint correction for two molecules generates a correct, overlapping distribution.
+    u_lhs_fn, u_rhs_fn, core_idxs, core_params, mol_a, mol_b = setup_system()
+
+    combined_mass = np.concatenate([
+        [a.GetMass() for a in mol_a.GetAtoms()],
+        [b.GetMass() for b in mol_b.GetAtoms()]
+    ])
+
+    combined_conf = np.concatenate([
+        get_romol_conf(mol_a),
+        get_romol_conf(mol_b)
+    ])
+
+    dt = 1.5e-3
+    friction = 1.0
+
+    temperature = 300.0
+
+    beta = 1/(constants.BOLTZ*temperature)
+
+    ca, cb, cc = langevin_coefficients(temperature, dt, friction, combined_mass)
+    cb = -1*np.expand_dims(cb, axis=-1)
+    cc = np.expand_dims(cc, axis=-1)
+
+    x_t = combined_conf
+    v_t = np.zeros_like(x_t)
+
+    n_steps = 50000
+    equilibrium_steps = 5000
+    sampling_frequency = 100
+
+    lhs_xs = []
+    lhs_du_dx_fn = jax.jit(jax.grad(u_lhs_fn))
+    for step in range(n_steps):
+        du_dx = lhs_du_dx_fn(x_t)
+        v_t = ca*v_t + cb*du_dx + cc*np.random.normal(size=x_t.shape)
+        x_t = x_t + v_t*dt
+        if step % sampling_frequency == 0 and step > equilibrium_steps:
+            lhs_xs.append(x_t)
+
+    # reset x and v
+    x_t = combined_conf
+    v_t = np.zeros_like(x_t)
+
+    rhs_xs = []
+    rhs_du_dx_fn = jax.jit(jax.grad(u_rhs_fn))
+    for step in range(n_steps):
+        du_dx = rhs_du_dx_fn(x_t)
+        v_t = ca*v_t + cb*du_dx + cc*np.random.normal(size=x_t.shape)
+        x_t = x_t + v_t*dt
+        if step % sampling_frequency == 0 and step > equilibrium_steps:
+            rhs_xs.append(x_t)
+
+    lhs_xs = np.array(lhs_xs)
+    rhs_xs = np.array(rhs_xs)
+
+    k_translation = 200.0
+
+    results = []
+    for k_rotation in [0.0, 50.0, 1000.0]:
```

Comment:
I think we can expand this later on to brute force over a grid of k_translation/k_rotation values to try and find the optimal overlap.